cvmfs/directory_entry.h

```diff
@@ -195,6 +196,31 @@ class DirectoryEntryBase {
     return s;
   }
 
+  /**
+   * Converts to a cvmfs_stat struct. This provides the
+   * information provided by stat, but also CVMFS specific
+   * information such as hash.
+   * @return the struct cvmfs_stat for this DirectoryEntry
+   */
+  inline void GetCVMFSStatStructure(struct cvmfs_attr *attr) const {
```

Comment:
I think we should move this to libcvmfs, and convert from the other side, like `cvmfs_attr::FromDirent`